elastic/apm-server
# Java attacher runs only in non-cloud environments

## Motivation/summary

start the java attacher jar only when running in a non-cloud environment

## Checklist


## How to test these changes

1. start apm-server with the java attacher config enabled and  `ELASTIC_AGENT_CLOUD` set in the environment
2. verify the java attacher does not start
3. start apm-server with the java attacher config enabled and  `ELASTIC_AGENT_CLOUD` NOT set in the environment
4. verify the java attacher does start

## Related issues

requested in https://github.com/elastic/apm-server/issues/5479
depends on https://github.com/elastic/apm-server/pull/6593
